boards/xtensa/odroid_go/doc/index.rst

```diff
@@ -0,0 +1,139 @@
++-------+------------------+-------------------------+
+
+Programming
+***********
+
+Two components are required in order to build applications for this board: the
+`toolchain`_ and the `SDK`_.  Both are provided by the SoC manufacturer.
+
+The SDK contains headers and a hardware abstraction layer library (provided only
+as object files) that are required for the port to function.
+
+The toolchain is available for Linux, Windows, and Mac hosts and instructions to
+obtain and set them up are available in the ESP-IDF repository, using the
+toolchain and SDK links above.
+
+Set up build environment
+------------------------
+
+With both the toolchain and SDK installed, the Zephyr build system must be
+instructed to use this particular variant by setting the following shell
+variables:
+
+.. code-block:: console
+
+   export ZEPHYR_TOOLCHAIN_VARIANT="espressif"
+   export ESPRESSIF_TOOLCHAIN_PATH="/path/to/xtensa-esp32-elf/"
+
+Flashing
+--------
+
+The usual ``flash`` target will work with the ``odroid_go`` board configuration.
+Here is an example for the :ref:`hello_world` application using CMake:
+
+.. code-block:: console
+
+        # From the root of the zephyr repository
+        # Use cmake to configure buildsystem:
+        cmake -DESP_IDF_PATH=/path/to/esp-idf/ -B build -DBOARD=odroid_go samples/hello_world
+
+        # Now run make to build the application:
+        make
+
+Refer to :ref:`build_an_application` and :ref:`application_run` for more
+details.
+
+All flashing options are now handled by the :ref:`west` tool, including flashing
+with custom options such as a different serial port.  The ``west`` tool supports
+specific options for the ESP32 board, as listed here:
+
+  --esp-idf-path ESP_IDF_PATH
+                        path to ESP-IDF
+  --esp-device ESP_DEVICE
+                        serial port to flash, default /dev/ttyUSB0
+  --esp-baud-rate ESP_BAUD_RATE
+                        serial baud rate, default 921600
+  --esp-flash-size ESP_FLASH_SIZE
+                        flash size, default "detect"
+  --esp-flash-freq ESP_FLASH_FREQ
+                        flash frequency, default "40m"
+  --esp-flash-mode ESP_FLASH_MODE
+                        flash mode, default "dio"
+  --esp-tool ESP_TOOL   if given, complete path to espidf. default is to
+                        search for it in [ESP_IDF_PATH]/components/esptool_py/
+                        esptool/esptool.py
+  --esp-flash-bootloader ESP_FLASH_BOOTLOADER
+                        Bootloader image to flash
+  --esp-flash-partition_table ESP_FLASH_PARTITION_TABLE
+                        Partition table to flash
+
+For example, to flash to ``/dev/ttyUSB0``, use the following command after
+having build the application in the ``build`` directory:
+
+.. code-block:: console
+
+   west flash -d build/ --skip-rebuild --esp-tool=/path/to/esp-idf/components/esptool_py/esptool/esptool.py --esp-device /dev/ttyUSB0
+
+References
+**********
+
```

Comment:
If you want the toolchain and SDK links (and not just the footnote) to show up here too, add the directive (followed by a blank line):
```
.. target-notes::

```